overlord/snapstate/snapstate.go

```diff
@@ -1512,6 +1519,23 @@ func updateManyFiltered(ctx context.Context, st *state.State, names []string, us
 	return updated, tasksets, nil
 }
 
+// filterHeldSnaps filters held snaps from being updated in a general refresh.
+func filterHeldSnaps(st *state.State, names []string, updates []minimalInstallInfo) ([]minimalInstallInfo, error) {
```

Comment:
the `names` parameter doesn't seem to be used

Reply:
My bad, it was used but I forgot to remove it when I refactored it out. Thanks